File: study/lua/src/app/scene/view/gachaGoldShop/GoldHeroShopItemCell.lua
-- Author: Panhao
-- Date: 

local ListViewCellBase = require("app.ui.ListViewCellBase")
local GoldHeroShopItemCell = class("GoldHeroShopItemCell", ListViewCellBase)
local TypeConvertHelper = require("app.utils.TypeConvertHelper")


function GoldHeroShopItemCell:ctor()
	self._itemInfo1 = nil
	self._itemInfo2 = nil

	local resource = {
		file = Path.getCSB("GoldHeroShopItemCell", "gachaGoldShop"),

	}
	GoldHeroShopItemCell.super.ctor(self, resource)
end

function GoldHeroShopItemCell:onCreate()
	local size = self._resourceNode:getContentSize()
	self:setContentSize(size.width, size.height)
	self:_initResourceView()
end

function GoldHeroShopItemCell:_initResourceView( ... )
    for i=1, 2 do
		local item = self["_itemInfo"..i]
		if item then
			item:setVisible(false)
		end
	end
end

function GoldHeroShopItemCell:update(index, itemLine, tabIdx)
	for i=1, 2 do
		local item = self["_itemInfo"..i]
		if item then
			item:setVisible(false)
		end
	end

	for i , item in pairs(itemLine) do
		local itemInfo = self["_itemInfo"..i]
        if itemInfo then
            itemInfo:updateUI(item, tabIdx == 1)
			itemInfo:setVisible(true)
			itemInfo:setTag(i + index*2)
            itemInfo:setCallBack(handler(self,self._onBtnClick))
            local discount = (item:getConfig().discount or 0)
            itemInfo:updateDiscount(tonumber(discount))
		end
	end
end

function GoldHeroShopItemCell:_onBtnClick(iconNameNode)
	local curSelectedPos = iconNameNode:getTag()
	self:dispatchCustomCallback(curSelectedPos)
end

function GoldHeroShopItemCell:findItemByPos(pos)
	for i=1 , 2  do
		local itemInfo = self["_itemInfo"..i]
		if itemInfo and itemInfo:getTag() == pos then
			return itemInfo
		end
	end
	return nil
end


return GoldHeroShopItemCell

File: study/ts/script/scene/view/gachaGoldShop/GoldHeroShopItemCell.ts
const {ccclass, property} = cc._decorator;

import CommonShopItemCell from '../../../ui/component/CommonShopItemCell'
import ListViewCellBase from '../../../ui/ListViewCellBase';
import { handler } from '../../../utils/handler';

@ccclass
export default class GoldHeroShopItemCell extends ListViewCellBase {

    @property({
        type: cc.Node,
        visible: true
    })
    _resourceNode: cc.Node = null;

    @property({
        type: CommonShopItemCell,
        visible: true
    })
    _itemInfo1: CommonShopItemCell = null;

    @property({
        type: CommonShopItemCell,
        visible: true
    })
    _itemInfo2: CommonShopItemCell = null;

    onInit(){
        var size = this._resourceNode.getContentSize();
        this.node.setContentSize(size.width, size.height);
    }
    onCreate() {
        this._initResourceView();
    }
    _initResourceView() {
        for (var i = 1; i<=2; i++) {
            var item = this['_itemInfo' + i];
            if (item) {
                item.setVisible(false);
            }
        }
    }
    updateUI(index, itemLine:any[], tabIdx) {
        for (var i = 1; i<=2; i++) {
            var item = this['_itemInfo' + i];
            if (item) {
                item.setVisible(false);
            }
        }
        for (var i=1; i<=itemLine.length; i++) {
            var item = itemLine[i-1];
            var itemInfo = this['_itemInfo' + i];
            if (itemInfo) {
                itemInfo.updateUI(item, tabIdx == 1);
                itemInfo.setVisible(true);
                itemInfo.node.name = (i + index * 2).toString();
                itemInfo.setCallBack(handler(this, this._onBtnClick));
                var discount = item.getConfig().discount || 0;
                itemInfo.updateDiscount(parseInt(discount));
            }
        }
    }
    _onBtnClick(iconNameNode) {
        var curSelectedPos = parseInt(iconNameNode.node.name);
        this.dispatchCustomCallback(curSelectedPos);
    }
    findItemByPos(pos) {
        for (var i = 1; i<=2; i++) {
            var itemInfo = this['_itemInfo' + i];
            var index = parseInt(itemInfo.node.name);
            if (itemInfo && index == pos) {
                return itemInfo;
            }
        }
        return null;
    }    

}
